src/stancjs/stancjs.ml

```diff
@@ -190,102 +197,132 @@ let wrap_result ?printed_filename ~code ~warnings res =
-    | BadJsInput s -> Some s
-    | _ -> None)
+let bad_arg_message ~name ~expected value =
+  Fmt.str
+    "Failed to convert stanc.js argument '%s'!@ It had type '%s' instead of \
+     '%s'."
+    name
+    (Js.typeof value |> Js.to_string)
+    expected
 
-let typecheck e typ = String.equal (Js.to_string (Js.typeof e)) typ
+let checked_to_string ~name value =
+  if not (typecheck value "string") then
+    Error (bad_arg_message ~name ~expected:"string" value)
+  else Ok (Js.to_string value)
+
+let checked_to_array ~name value =
+  if
+    not
+      (Js.to_bool
+         (Js.Unsafe.meth_call
+            (Js.Unsafe.pure_js_expr "Array")
+            "isArray"
+            [| Js.Unsafe.inject value |]))
+  then
+    Error
+      (Fmt.str
+         "Failed to convert stanc.js argument '%s'!@ Array.isArray returned \
+          false for value of type '%s'."
+         name
+         (Js.typeof value |> Js.to_string))
+  else Ok (Js.to_array value)
 
 (** Converts from a [{ [s:string]:string }] JS object type
 to an OCaml map, with error messages on bad input. *)
 let to_file_map includes =
-  try
-    match Js.Optdef.to_option includes with
-    | None -> Result.Ok String.Map.empty (* normal use: argument not supplied *)
-    | Some includes ->
+  let open Result_let in
+  match Js.Opt.to_option includes with
+  | None -> Result.Ok String.Map.empty (* normal use: argument not supplied *)
+  | Some includes ->
+      let* () =
         if not (typecheck includes "object") then
-          raise
-            (BadJsInput
-               "Included files map was provided but was not of type 'object'");
-        let keys = Js.object_keys includes |> Js.to_array |> List.of_array in
-        let value k =
-          let value_js = Js.Unsafe.get includes k in
-          if typecheck value_js "string" then
-            value_js |> Js.Unsafe.coerce |> Js.to_string
-          else
-            raise
-              (BadJsInput
-                 (Fmt.str
-                    "Failed to read property '%s' of included files map!@ It \
-                     had type '%s' instead of 'string'."
-                    (Js.to_string k)
-                    (Js.typeof value_js |> Js.to_string))) in
-        Result.Ok
-          (String.Map.of_alist_exn (* JS objects cannot have duplicate keys *)
-             (List.map keys ~f:(fun k ->
-                  let key_clean = k |> Js.to_string in
-                  (key_clean, value k))))
-  with
-  | BadJsInput s -> Result.Error s
-  | e -> Result.Error (Exn.to_string e)
+          Error (bad_arg_message ~name:"includes" ~expected:"object" includes)
+        else Ok () in
+      let keys = Js.object_keys includes |> Js.to_array |> List.of_array in
+      let value k =
+        let value_js = Js.Unsafe.get includes k in
+        checked_to_string
+          ~name:("includes[\"" ^ Js.to_string k ^ "\"]")
+          value_js in
+      let* values = List.map keys ~f:value |> Result.all in
```

Comment:
`Result.all` discards all includes if the object contains any non-string values. This error is not considered fatal so I think you should use `List.partition_result` instead.